notty-cairo/src/lib.rs

```diff
@@ -59,10 +56,16 @@ impl Renderer {
                             text = TextRenderer::new(x_pixels(canvas, x_pos + width as usize),
                                                      y_pix);
                         }
+                        if let Some(image) = self.images.borrow().get(data) {
+                            image.draw(canvas);
+                            continue;
+                        }
                         let x_pix = x_pixels(canvas, x_pos);
                         let w_pix = x_pixels(canvas, width as usize);
                         let h_pix = y_pixels(canvas, height as usize);
-                        ImageRenderer::new(&data, x_pix, y_pix, w_pix, h_pix, pos).draw(canvas);
+                        let img = ImageRenderer::new(data, x_pix, y_pix, w_pix, h_pix, pos);
+                        img.draw(canvas);
+                        self.images.borrow_mut().insert(data.to_owned(), img);
```

Comment:
Yes, if images are no longer accessible from the Terminal struct they should expire, currently they'll take up memory in this hashmap forever.

There's a way to do this I think using `Rc`s and weak pointers, but I'm not sure exactly what it should look like.